### ISSUE TITLE
AlterColumn crashes with System.IndexOutOfRangeException if run with PreviewOnly=true

### ISSUE THREAD

**Issue by dpsenner** [fluentmigrator/fluentmigrator#817]
These migrations:

```
    [Migration(100)]
    public class Migration_v100_AddUsersTable : Migration
    {
        public override void Up()
        {
            Create.Table("Users")
                .WithColumn("Id")
                    .AsFloat()
                    .NotNullable()
                    .PrimaryKey()
                    .Identity()
                .WithColumn("Name")
                    .AsString(60)
                    .NotNullable()
                    .WithDefaultValue("Anonymous");
        }

        public override void Down()
        {
            Delete.Table("Users");
        }
    }
    [Migration(200, "Patch the user table such that the length of the name column is increased from 60 to 80 characters.")]
    public class Migration_v200_AlterTableUsersModifyColumnName : Migration
    {
        public override void Up()
        {
            Alter.Table("Users")
                .AlterColumn("Name")
                .AsString(80);
        }

        public override void Down()
        {
            throw new NotImplementedException();
        }
    }
```

Cause this exception when migrating on a firebird database in preview mode:

```
-- 200: Migration_v200_AlterTableUsersModifyColumnName migrating
-- Beginning Transaction
-- AlterTable Users

-- elapsed: 00:00:00.0020004
-- AlterColumn Users Name String
-- ERROR: System.IndexOutOfRangeException: There is no row at position 0.
   at System.Data.RBTree`1.GetNodeByIndex(Int32 userIndex)
   at FluentMigrator.Runner.Processors.Firebird.TableInfo.Read(FirebirdProcessor processor, String tableName) in C:\Users\Mark\Source\Repos\fluentmigrator\src\FluentMigrator.Runner\Processors\Firebird\FirebirdSchemaInfo.cs:line 44
   at FluentMigrator.Runner.Processors.Firebird.FirebirdTableSchema.LoadMeta() in C:\Users\Mark\Source\Repos\fluentmigrator\src\FluentMigrator.Runner\Processors\Firebird\FirebirdTableSchema.cs:line 45
   at FluentMigrator.Runner.Processors.Firebird.FirebirdTableSchema.Load() in C:\Users\Mark\Source\Repos\fluentmigrator\src\FluentMigrator.Runner\Processors\Firebird\FirebirdTableSchema.cs:line 36
   at FluentMigrator.Runner.Processors.Firebird.FirebirdTableSchema..ctor(String tableName, FirebirdProcessor processor) in C:\Users\Mark\Source\Repos\fluentmigrator\src\FluentMigrator.Runner\Processors\Firebird\FirebirdTableSchema.cs:line 31
   at FluentMigrator.Runner.Processors.Firebird.FirebirdSchemaProvider.LoadTableSchema(String tableName) in C:\Users\Mark\Source\Repos\fluentmigrator\src\FluentMigrator.Runner\Processors\Firebird\FirebirdSchemaProvider.cs:line 37
   at FluentMigrator.Runner.Processors.Firebird.FirebirdSchemaProvider.GetTableSchema(String tableName) in C:\Users\Mark\Source\Repos\fluentmigrator\src\FluentMigrator.Runner\Processors\Firebird\FirebirdSchemaProvider.cs:line 31
   at FluentMigrator.Runner.Processors.Firebird.FirebirdProcessor.Process(AlterColumnExpression expression) in C:\Users\Mark\Source\Repos\fluentmigrator\src\FluentMigrator.Runner\Processors\Firebird\FirebirdProcessor.cs:line 388
   at FluentMigrator.Expressions.AlterColumnExpression.ExecuteWith(IMigrationProcessor processor) in C:\Users\Mark\Source\Repos\fluentmigrator\src\FluentMigrator\Expressions\AlterColumnExpression.cs:line 48
   at FluentMigrator.Runner.MigrationRunner.<>c__DisplayClass72_0.<ExecuteExpressions>b__1() in C:\Users\Mark\Source\Repos\fluentmigrator\src\FluentMigrator.Runner\MigrationRunner.cs:line 489
   at FluentMigrator.Runner.StopWatch.Time(Action action) in C:\Users\Mark\Source\Repos\fluentmigrator\src\FluentMigrator.Runner\StopWatch.cs:line 57
   at FluentMigrator.Runner.MigrationRunner.ExecuteExpressions(ICollection`1 expressions) in C:\Users\Mark\Source\Repos\fluentmigrator\src\FluentMigrator.Runner\MigrationRunner.cs:line 489
```

The problem appears obvious: there is no such table named "Users" in the database to read the table info from because migration 100 has not been run yet.

**Comment by dpsenner** [comment]
This relates to #674 because the stacktrace is very similar.

**Comment by fubar-coder** [comment]
The Firebird processor tries to get/use the current table definition, but it couldn't find one (migration 100 not run yet). Fixed in release/2.0.0
